runelite-client/src/main/java/net/runelite/client/plugins/blastmine/BlastMineRockOverlay.java

```diff
@@ -0,0 +1,196 @@
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.blastmine;
+
+import com.google.common.collect.Sets;
+import net.runelite.api.*;
+import net.runelite.api.coords.LocalPoint;
+import net.runelite.api.widgets.Widget;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.components.ProgressPie;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.Polygon;
+import java.awt.image.BufferedImage;
+import java.util.Collections;
+import java.util.Set;
+
+public class BlastMineRockOverlay extends Overlay
+{
+	private static final int TIMER_SIZE = 25;
+	private static final int TIMER_BORDER_WIDTH = 1;
+	private static final int MAX_DISTANCE = 16;
+	private static final int WARNING_DISTANCE = 2;
+	private static final Set<Integer> WALL_OBJECTS = Collections.unmodifiableSet(Sets.newHashSet(ObjectID.NULL_28570, ObjectID.NULL_28571, ObjectID.NULL_28572, ObjectID.NULL_28573, ObjectID.NULL_28574,
+			ObjectID.NULL_28575, ObjectID.NULL_28576, ObjectID.NULL_28577, ObjectID.NULL_28578, ObjectID.HARD_ROCK, ObjectID.HARD_ROCK_28580,
+			ObjectID.CAVITY, ObjectID.CAVITY_28582, ObjectID.POT_OF_DYNAMITE, ObjectID.POT_OF_DYNAMITE_28584, ObjectID.POT_OF_DYNAMITE_28585,
+			ObjectID.POT_OF_DYNAMITE_28586, ObjectID.SHATTERED_ROCKFACE, ObjectID.SHATTERED_ROCKFACE_28588));
+
+	private final Client client;
+	private final BlastMinePlugin plugin;
+	private final BlastMinePluginConfig config;
+	private final ItemManager itemManager;
+
+	private Color timerColor;
+	private Color timerBorderColor;
+	private Color warningColor;
+
+	@Inject
+	BlastMineRockOverlay(@Nullable Client client, BlastMinePlugin plugin, BlastMinePluginConfig config, ItemManager itemManager)
```

Comment:
Client is not nullable unless the plugin is configured to load when outdated, which it should not be.